Source/ActionController.swift

```diff
@@ -265,340 +266,340 @@ public class ActionController<ActionViewType: UICollectionViewCell, ActionDataTy
                     return cancel
                 }()
             }
             view.addSubview(cancelView!)
         }
     }
     
-    public override func viewWillAppear(animated: Bool) {
+    open override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         backgroundView.frame = view.bounds
     }
 
-    public override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
-        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+    open override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
         
         if let _ = settings.animation.scale {
-            presentingViewController?.view.transform = CGAffineTransformIdentity
+            presentingViewController?.view.transform = CGAffineTransform.identity
         }
         
         self.collectionView.collectionViewLayout.invalidateLayout()
         
-        coordinator.animateAlongsideTransition({ [weak self] _ in
+        coordinator.animate(alongsideTransition: { [weak self] _ in
             self?.setUpContentInsetForHeight(size.height)
             self?.collectionView.reloadData()
             if let scale = self?.settings.animation.scale {
-                self?.presentingViewController?.view.transform = CGAffineTransformMakeScale(scale.width, scale.height)
+                self?.presentingViewController?.view.transform = CGAffineTransform(scaleX: scale.width, y: scale.height)
             }
         }, completion: nil)
     }
 
-    override public func viewDidLayoutSubviews() {
+    override open func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()
         collectionView.collectionViewLayout.invalidateLayout()
     }
     
-    public override func prefersStatusBarHidden() -> Bool {
+    open override var prefersStatusBarHidden : Bool {
         return !settings.statusBar.showStatusBar
     }
     
-    public  override func preferredStatusBarStyle() -> UIStatusBarStyle {
+    open  override var preferredStatusBarStyle : UIStatusBarStyle {
         return settings.statusBar.style
     }
         
     // MARK: - UICollectionViewDataSource
     
-    public func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
+    open func numberOfSections(in collectionView: UICollectionView) -> Int {
         return numberOfSections()
     }
     
-    public func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    open func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         if hasHeader() && section == 0 { return 0 }
-        if settings.behavior.useDynamics && section > _dynamicSectionIndex {
+        if settings.behavior.useDynamics && section > _dynamicSectionIndex ?? 0 {
```

Comment:
This is not exactly the same, now if `_dynamicSectionIndex` is null and `section` is 0, then this function si returning `_sections[actionSectionIndexFor(section)].actions.count`. While previously it returned 0.

Additionally, this is causing the crash that you saw. When `_dynamicSectionIndex` is null, it means that no data has been added to the collection view (in the case `useDynamics==true`). So the collection view datasource is returning x number of rows for section 0 when no one has been inserted.

Next code should be fix the crash:
```swift
    open func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        if hasHeader() && section == 0 { return 0 }
        let rows = _sections[actionSectionIndexFor(section)].actions.count

        guard let dynamicSectionIndex = _dynamicSectionIndex else {
            return settings.behavior.useDynamics ? 0 : rows
        }
        if settings.behavior.useDynamics && section > dynamicSectionIndex {
            return 0
        }
        return rows
    }
```